packages/components/src/form-file-upload/index.js

```diff
@@ -13,6 +13,7 @@ function FormFileUpload( {
 	children,
 	multiple = false,
 	onChange,
+	onInputFileClick,
```

Comment:
Thanks for your feedback, I didn't name it `onClick` because I thought it would be like a click on the entire component and not only the `<input type="file">` inside it, but that's a good point, a developer would probably use `onClick` first, I'll do the modifications.

Reply:
`onClick` renamed on 326e3c6 and added UI tests on c839698, needed to add the `@testing-library/user-event` package.

Reply:
We recently discussed the introduction of `@testing-library/user-event` and came to the conclusion that we should [introduce it as a new dependency in a separate PR ](https://github.com/WordPress/gutenberg/pull/39148#issuecomment-1056642675).

I'll try to give this a priority and open a PR later today — once it gets merged, we'll be able to rebase this PR on top of `trunk`.

Reply:
For reference, #39360

Reply:
Fixed in e9d9419 , I had some problems with the rebase command (used `git pull --rebase` instead of `git push -f`) that requested review of the latest committers.